### PR TITLE
webpack resolver: Simplify loop

### PR DESCRIPTION
**Pull request by fregante**
This PR reduces the amount of code run by using a plain for loop with early return. I was hesitant to make the change at first but I noticed that there are a lot of similar loops in the code so it shouldn't be a problem to use the same here

### CODE REVIEW

File: resolvers/webpack/index.js [hunk 1]
```diff
@@ -293,17 +292,20 @@ const MAX_CACHE = 10;
...
-  let cached = find(_cache, function (entry) { return isEqual(entry.key, cacheKey); });
-  if (!cached) {
-    cached = {
-      key: cacheKey,
-      value: createResolveSync(configPath, webpackConfig, cwd),
-    };
-    // put in front and pop last item
-    if (_cache.unshift(cached) > MAX_CACHE) {
-      _cache.pop();
+  for (const entry of _cache) {
```
Comment (ljharb) [inline resolvers/webpack/index.js:295]: this uses for-of tho, which is slower than standard array iteration - a standard for loop would be the replacement here

Comment (fregante) [reply resolvers/webpack/index.js:295]: 2c577cb  can be undone if plain `for` loops are preferable, but I see both `for-of` and plain `for` loops used everywhere in the repo. I'd say `for-of` gives a good balance between performance (still better than polyfilling .find()) and readability (over plain `for` loops)

Comment (fregante) [reply resolvers/webpack/index.js:295]: Let me know if I should revert `2c577cb` or if for-of is good enough here

Comment (fregante) [reply resolvers/webpack/index.js]: Done, it uses a regular for loop now

**Review by ljharb — APPROVED** [review]

**Pull request merged**